php/Terminus/Models/Collections/TerminusCollection.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Terminus\Models\Collections;
+
+use \TerminusCommand;
+
+abstract class TerminusCollection {
+  protected $fetch_args = array();
+  protected $fetch_url  = '';
+  protected $models     = array();
+  protected $owner_name = false;
+
+  /**
+   * Instantiates the collection, sets param members as properties
+   *
+   * @param [array] $options To be set to $this->key
+   * @return [TerminusCollection] $this
+   */
+  public function __construct($options = array()) {
+    foreach ($options as $key => $option) {
+      $this->$key = $option;
+    }
+  }
+
+  /**
+   * Retrieves all models
+   *
+   * @return [array] $this->models
+   */
+  public function all() {
+    if (empty($this->models)) {
+      $this->fetch();
+    }
+    return $this->models;
```

Comment:
Rather than returning the property, can we dupe it and only return array_keys? I'd like the models property to be protected, and have the return values of accessors encourage functional usage.